### ISSUE TITLE
`non_fmt_panics` is trigged when `panic!` is declared in an external crate

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust#96997]
Tested on the `master` branch. Feel free to close this issue if duplicated.

```rust
// External crate. Let's call it "foo"

#[macro_export]
macro_rules! bad {
  ($msg:literal) => {
    panic!($msg)
  }
}
```

```rust
// Another crate

fn _bar() {
  macro_rules! good {
    ($msg:literal) => {
      panic!($msg)
    }
  }

  foo::bad!("ABC {{ BCA"); // Lint is triggered but shouldn't because `{` is escaped
  good!("ABC {{ BCA"); // Does not trigger as expected
}
```

Maybe some encoding issue between boundaries ?

**Comment by c410-f3r** [comment]
Closing due to the lack of activeness
